### PR TITLE
[dev] Optimize unnecessary workflow runs

### PR DESCRIPTION
**Pull request by naman03malhotra**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
Optimize unnecessary workflow runs for the cherry-pick workflow. Currently it is only being used when there is milestone change in the PR, however, it is also being triggered when there is a label change, creating unnecessary runs and causing delays for other runs.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Fork this repo.
2. Create a milestone `9.5.0` if you don't already have.
3. Hard reset the trunk branch with `dev/update-cherry-pick-flow`
4. `git reset --hard [origin]/dev/update-cherry-pick-flow`
5. Create a PR against the trunk.
6. Change labels.
7. Note that the action shouldn't trigger.
8. Now merge the PR.
9. Not that the action shouldn't trigger.
10. Now change the milestone for the PR to let's say, `9.5.0`
11. The workflow should trigger, but it might fail if you don't have `release/9.5` and that's okay, we are only testing the trigger.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/cherry-pick.yml
Comment (coreymckrill) [inline .github/workflows/cherry-pick.yml:4]: Shouldn't the `types: [milestoned]` be for `pull_request` instead of `issues`?

**Review by coreymckrill — COMMENTED** [review]
I got to step 9 in the test instructions, but it appears that the Cherry Pick Tool workflow ran when I merged the PR.

image

I'm not sure if I did everything right though. I do think that someone from @woocommerce/flux should also review this.

File: .github/workflows/cherry-pick.yml
Comment (naman03malhotra) [reply .github/workflows/cherry-pick.yml:4]: So I was also under the same assumption, but the milestone is only supported under issues, and github treats PRs are issue under the hood for some cases.

**Comment by naman03malhotra** [comment]
>but it appears that the Cherry Pick Tool workflow ran when I merged the PR.

Those in the screenshots are the new flows, its fine for them to run.

We need to observe https://github.com/woocommerce/woocommerce/blob/trunk/.github/workflows/cherry-pick.yml

They have similar names xD

**Comment by naman03malhotra** [comment]
@kalessil I've assigned it you to take another look.

**Review by kalessil — APPROVED** [review]
LGTM (code review only)!

**Comment by coreymckrill** [comment]
Closing / re-opening to jumpstart stalled CI

**Comment by coreymckrill** [comment]
@naman03malhotra since 9.5 is out now, do we need to update the target branch on this to 9.6?

**Comment by naman03malhotra** [comment]
>since 9.5 is out now, do we need to update the target branch on this to 9.6?

I've intentionally targeted 9.5 because we also have an upcoming security release, and it is good to have it on that branch, too.

**Comment by naman03malhotra** [comment]
I assigned it to @jorgeatorres as well, incase Corey couldn't get to it before AFK.

**Review by coreymckrill — APPROVED** [review]
Ok, after the merge I changed the milestone, and the Cherry Pick Tool ran, as expected.

image

(And failed, as expected, since my fork didn't have the right branch.)

**Comment by coreymckrill** [comment]
Now it appears that the workflows that are introduced in this PR are failing for this PR and blocking merge. Do we need to add a milestone here? @naman03malhotra @jorgeatorres

**Pull request merged**
